var/spack/repos/builtin/packages/htslib/package.py

```diff
@@ -49,6 +49,11 @@ class Htslib(AutotoolsPackage):
         default=True,
         description="use libdeflate for faster crc and deflate algorithms",
     )
+    variant("gcs", default=False, description="enable gcs url support")
```

Comment:
So this comes from Rob and James more than me:

- year2038  Only needed on 32-bit platforms. Does spack bother with those?  
- pic is only needed if an external package is building a shared library, and tries to link in the static libhts.a file.  If instead it links against libhts.so it will find life much easier as it doesn't have to deal with HTSlib's dependencies.  While mostly harmless, I expect most packages that link to libhts.a also provide their own copy of HTSlib and so will manage HTSlib's build options themselves. 
- The plugins one might be useful 
- S3 and gcs should I think default to true if you also enable curl . Note that explicitly using --enable-gcs and --enable-S3 will fail if you don't also --enable-libcurl . Explicitly disabling them might be useful if you want a more reproducible build as otherwise they will search around for the required libraries and may find them on the system (the same is true for libcurl).  S3 should add openssl as a dependency to get libcrypto .